File: Jardin De infantes/UI/Servicios/Idioma/iuCambiarIdioma.vb
Imports BLL
Imports EE

Imports System.Globalization
Imports System.Threading
Public Class iuCambiarIdioma
    Sub cambiaridioma()

        Dim unbllTraducir As New bllTraduccion


        Dim unTraducir As New Traduccion(SingletonIdioma.intance.getUsuario)
        Dim ht As New Hashtable
        ht = unbllTraducir.ListarPalabrasTraducidas(unTraducir)
        For Each item As DictionaryEntry In ht
            For Each controlcito As Control In Me.Controls
                If controlcito.Tag Is Nothing Then
                Else
                    If controlcito.Tag.Equals(item.Key) Then
                        controlcito.Text = item.Value

                    End If
                End If

            Next
        Next
        Dim culturaObj As CultureInfo = New CultureInfo(SingletonIdioma.intance.getUsuario.getIdioma)
        Threading.Thread.CurrentThread.CurrentUICulture = culturaObj
        Threading.Thread.CurrentThread.CurrentCulture = culturaObj
    End Sub
    Private Sub MetroComboBox1_SelectedIndexChanged(sender As Object, e As EventArgs) Handles cbIdioma.SelectedIndexChanged

    End Sub

    Private Sub iuCambiarIdioma_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Dim unBll As New bllIdioma
        cbIdioma.DataSource = unBll.listaIdiomaString
        cambiaridioma()
    End Sub

    Private Sub btnModificar_Click(sender As Object, e As EventArgs) Handles btnModificar.Click
        SingletonIdioma.intance.CambiarIdioma(cbIdioma.Text)
        cambiaridioma()
        Me.Close()
    End Sub
End Class

File: Jardin De infantes/Seguridad/IU/iuCambiarIdioma.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using BLLseguridad.Idioma;

namespace IU
{
    public partial class iuCambiarIdioma : MetroFramework.Forms.MetroForm
    {
        void CambiarCultura()
        {
            CultureInfo cultura = new CultureInfo(EEseguridad.Idioma.SingletonIdioma.intance.getUsuario().idioma);
            System.Threading.Thread.CurrentThread.CurrentUICulture = cultura;
            System.Threading.Thread.CurrentThread.CurrentCulture = cultura;

        }
        void CambiarIdioma()
        {
            bllTraduccion bll = new bllTraduccion();


            
            bllTraduccion traduccion = new bllTraduccion();
            Hashtable ht = new Hashtable();
            ht = traduccion.ObtenerTraduccion(EEseguridad.Idioma.SingletonIdioma.intance.getUsuario().idioma);
            foreach (DictionaryEntry item in ht)
            {
                foreach (Control controles in this.Controls)
                {
                    if (controles.Tag == null)
                    {

                    }
                    else
                    {
                        if (controles.Tag.Equals(item.Key))
                        {
                            controles.Text = item.Value.ToString();
                        }
                    }
                }
            }


        }



        public iuCambiarIdioma()
        {
            InitializeComponent();
        }

        private void iuCambiarIdioma_Load(object sender, EventArgs e)
        {
            helpProvider1.HelpNamespace = CHM.Cadena;
            bllIdioma bll = new bllIdioma();
            DataTable dt = new DataTable();
            dt = bll.ListarIdiomas();
            txtIdioma.DataSource = dt;
            txtIdioma.DisplayMember = "Idioma";
            txtIdioma.ValueMember = "Idioma";
            CambiarIdioma();
            CambiarCultura();
        }

        private void btnCambiarIdioma_Click(object sender, EventArgs e)
        {
            EEseguridad.Idioma.SingletonIdioma.intance.CambiarIdioma(txtIdioma.Text);
            CambiarIdioma();
            CambiarCultura();
        }
    }
}
